### ISSUE TITLE
Auto detection of GKE resource 

### ISSUE THREAD

**Issue by dukeng** [open-telemetry/opentelemetry-go-contrib#49]
OpenTelemetry should detect GKE openCensus resources and send them to the desired collector/exporter.

OC has a [resource mapping](https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/master/RESOURCE.md#k8s_container). We should consider the same feature for OT.

**Comment by nilebox** [comment]
Possibly a duplicate of / overlap with https://github.com/open-telemetry/opentelemetry-go/issues/785

**Comment by Aneurysm9** [comment]
I moved this issue to -contrib because it asked for resource detection for a specific vendor product, which we generally would not put in the main repo.  I think the designed proposed for https://github.com/open-telemetry/opentelemetry-go/issues/785 allows for additional resource detectors and the -contrib repo would be a good place for GKE/EKS/AKS detectors that go beyond the bounds of common k8s attributes.

**Comment by MrAlias** [comment]
Blocked on: https://github.com/open-telemetry/oteps/pull/111

**Comment by MrAlias** [comment]
Resolved by https://github.com/open-telemetry/opentelemetry-go-contrib/pull/154
